### PR TITLE
feat(abstract-utxo): add address creation for descriptor wallets 

### PR DESCRIPTION
**Pull request by ericli-bitgo**

### CODE REVIEW

File: modules/abstract-utxo/src/abstractUtxoCoin.ts [hunk 1]
```diff
@@ -231,26 +233,48 @@ export interface ParsedTransaction<TNumber extends number | bigint = number> ext
...
+  /** Descriptor value */
+  value: string;
+  /** Descriptor signatures */
+  signatures: string[];
+  /** Highest address index */
+  lastIndex: number;
+};
+
+export interface GenerateDescriptorAddressOptions extends GenerateAddressOptions {
+  descriptor: NamedDescriptor;
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/abstractUtxoCoin.ts:262]:
```suggestion
  descriptors: NamedDescriptor[];
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/abstract-utxo/src/abstractUtxoCoin.ts [hunk 2]
```diff
@@ -1052,6 +1084,12 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
     return [KeyIndices.USER, KeyIndices.BACKUP, KeyIndices.BITGO];
   }
 
+  isDescriptorWallet(
+    params: GenerateFixedAddressOptions | GenerateDescriptorAddressOptions | VerifyAddressOptions
+  ): params is GenerateDescriptorAddressOptions {
+    return 'descriptor' in params;
+  }
+
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/abstractUtxoCoin.ts:1092]: this should be a static method

File: modules/abstract-utxo/src/abstractUtxoCoin.ts [hunk 3]
```diff
@@ -997,23 +1021,31 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
...
 
-    if ((_.isUndefined(chain) && _.isUndefined(index)) || !(_.isFinite(chain) && _.isFinite(index))) {
-      throw new InvalidAddressDerivationPropertyError(
-        `address validation failure: invalid chain (${chain}) or index (${index})`
-      );
-    }
+    let expectedAddress: AddressDetails;
+    if (params.coinSpecific && 'descriptor' in params.coinSpecific) {
+      expectedAddress = this.generateAddress({
+        descriptor: params.coinSpecific.descriptor,
```
Comment (OttoAllmendinger) [inline modules/abstract-utxo/src/abstractUtxoCoin.ts:1027]: we only submit the name of the descriptor (the full descriptor itself is defined on the wallet)

**Review by pranavjain97 — COMMENTED** [review]
WP changes look fine. Please request re-review when Otto's changes are addressed

**Review by OttoAllmendinger — APPROVED** [review]

**Pull request merged**